pkg/security/utils/proc_linux.go

```diff
@@ -468,3 +468,101 @@ func FindPidNamespace(nspid uint32, ns uint64) (uint32, error) {
 	}
 	return 0, errors.New("PID not found")
 }
+
+// GetTracerPid returns the tracer pid of the the givent root pid
+func GetTracerPid(pid uint32) (uint32, error) {
+	statusFile := StatusPath(pid)
+	content, err := os.ReadFile(statusFile)
+	if err != nil {
+		return 0, fmt.Errorf("failed to read status file: %w", err)
+	}
+
+	lines := strings.Split(string(content), "\n")
+	for _, line := range lines {
+		if strings.HasPrefix(line, "TracerPid:") {
+			// Remove "NSpid:" prefix and trim spaces
+			line = strings.TrimPrefix(line, "TracerPid:")
+			line = strings.TrimSpace(line)
+
+			tracerPid, err := strconv.ParseUint(line, 10, 64)
+			if err != nil {
+				return 0, fmt.Errorf("failed to parse TracerPid value: %w", err)
+			}
+			return uint32(tracerPid), nil
+		}
+	}
+	return 0, fmt.Errorf("TracerPid field not found")
+}
+
+// FindTraceesByTracerPid returns the process list being trced by the given tracer host PID
+func FindTraceesByTracerPid(pid uint32) ([]uint32, error) {
+	procPids, err := process.Pids()
+	if err != nil {
+		return nil, err
+	}
+
+	traceePids := []uint32{}
+	for _, procPid := range procPids {
+		tracerPid, err := GetTracerPid(uint32(procPid))
+		if err != nil {
+			continue
+		}
+		if tracerPid == pid {
+			traceePids = append(traceePids, uint32(procPid))
+		}
+	}
+	return traceePids, nil
+}
+
+var isNsPidAvailable = sync.OnceValue(func() bool {
+	selfStatus := filepath.Join(kernel.ProcFSRoot(), "self", "status")
```

Comment:
I think in this case we might want the `/proc` without host in all cases no ?

Reply:
Hmm yes, let me update once more :)